brambling/views/payment.py

```diff
@@ -120,3 +124,89 @@ def post(self, request, *args, **kwargs):
         txn.is_confirmed = True if data['Transaction']['Status'] == 'processed' else False
         txn.save()
         return HttpResponse('')
+
+
+class StripeWebhookView(View):
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, *args, **kwargs):
+        return super(StripeWebhookView, self).dispatch(*args, **kwargs)
+
+    def post(self, request):
+        if request.META['CONTENT_TYPE'] != 'application/json':
+            raise Http404('Incorrect content type')
+
+        try:
+            event_data = json.loads(request.body)
+        except ValueError:
+            raise Http404('Webhook failed to decode request body')
+
+        stripe_event_id = event_data.get('id')
+        if not stripe_event_id:
+            raise Http404('Event does not have an id')
+
+        if event_data.get('livemode', False):
+            stripe_prep(LIVE)
+        else:
+            stripe_prep(TEST)
+
+        try:
+            event = stripe.Event.retrieve(stripe_event_id)
+        except stripe.error.InvalidRequestError:
+            raise Http404('Event not found on stripe')
+
+        if event.type != 'charge.refunded':
+            return HttpResponse(status=200)
+
+        with transaction.atomic():
+            try:
+                (ProcessedStripeEvent.objects
+                 .select_for_update().get(stripe_event_id=stripe_event_id))
+            except ProcessedStripeEvent.DoesNotExist:
+                ProcessedStripeEvent.objects.create(
+                    stripe_event_id=stripe_event_id)
+            else:
+                return HttpResponse(status=200)
+
+        try:
+            refund_id = event['data']['object']['refunds']['data'][0]['id']
+            refund = stripe.Refund.retrieve(refund_id, expand=['balance_transaction'])
+        except KeyError:
+            return HttpResponse(status=200)
+        except stripe.error.InvalidRequestError:
+            raise Http404('Refund not found on stripe')
+
+        try:
+            charge_id = event['data']['object']['id']
+            txn = Transaction.objects.get(remote_id=charge_id)
+        except Transaction.DoesNotExist:
+            return HttpResponse(status=200)
+        except KeyError:
+            raise Http404('Charge id not found')
+
+        try:
+            charge = stripe.Charge.retrieve(charge_id,
+                                            expand=['balance_transaction'])
+        except stripe.error.InvalidRequestError:
+            raise Http404('Charge not found on stripe')
+
+        try:
+            application_fee = stripe.ApplicationFee.all(charge=charge).data[0]
```

Comment:
I think we should probably handle the refund even if there are no associated application fees?